citro3d/src/uniform.rs

```diff
@@ -20,38 +22,147 @@ impl From<Index> for i32 {
     }
 }
 
-mod private {
-    use crate::math::Matrix;
+/// A uniform which may be bound as input to a shader program
+pub enum Uniform {
+    /// Single float uniform (`.fvec name`)
+    Float(FVec4),
+    /// Two element float uniform (`.fvec name[2]`)
+    Float2([FVec4; 2]),
```

Comment:
These would be good to `#[doc(alias)]` their C3D counterparts e.g. `C3D_FVUnifMtx2x4`. I guess we have no real equivalent for `C3D_FVUnifMtxNx4` so maybe that one could just be aliased on the struct as a whole?